drivers/Aqara/aqara-presence-sensor/src/fp2/device_manager.lua

```diff
@@ -0,0 +1,282 @@
+  ["14.55.85"] = { zone = "", event_handler = device_manager.monitoring_mode_handler },
+  ["4.31.85"] = { zone = "", event_handler = device_manager.fall_event_handler },
+  ["200.2.20000"] = { zone = "", event_handler = device_manager.zone_quantities_handler }
+}
+
+function device_manager.init_presence(driver, device)
+  if device:supports_capability(PresenceSensor)
+      and device:get_latest_state("main", PresenceSensor.ID, PresenceSensor.presence.NAME) == nil then
+    device:emit_event(PresenceSensor.presence("not present"))
+  end
+end
+
+function device_manager.init_movement(driver, device)
+  if not device:supports_capability(MovementSensor) then return end
+  device:emit_event(MovementSensor.movement("inactive"))
+end
+
+function device_manager.init_activity(driver, device)
+  if not device:supports_capability(ActivitySensor) then return end
+  device:emit_event(ActivitySensor.activity("noActivity"))
+end
+
+function device_manager.set_zone_info_to_latest_state(driver, device)
+  if not device:supports_capability(capabilities.multipleZonePresence) then return end
+
+  local zoneInfoTable = device:get_latest_state("main", capabilities.multipleZonePresence.ID,
+    capabilities.multipleZonePresence.zoneState.NAME, {})
+  multipleZonePresence.setZoneInfo(driver, device, zoneInfoTable)
+end
+
+function device_manager.handle_status(driver, device, status)
+  if not status then
+    log.warn("device_manager.handle_status : status is nil")
+    return
+  end
+
+  for k, _ in pairs(status) do
+    if resource_id[k] then
+      resource_id[k].event_handler(driver, device, resource_id[k].zone, tonumber(status[k]))
+    else
+      log.warn("device_manager.handle_status : resource id status is nil")
+    end
+  end
+end
+
+function device_manager.update_status(driver, device)
+  local conn_info = device:get_field(fields.CONN_INFO)
+
+  if not conn_info then
+    log.warn(string.format("device_manager.update_status : failed to find conn_info, dni = %s",
+      device.device_network_id))
+    return
+  end
+
+  local _, err, status = conn_info:get_attr()
+
+  if err or status ~= 200 then
+    log.error(string.format("device_manager.update_status : failed to get status, dni= %s, err= %s, status= %s",
+      device.device_network_id, err, status))
+    if status == 404 then
+      device:offline()
+    end
+    return
+  end
+end
+
+local sse_event_handlers = {
+  ["message"] = device_manager.handle_status
+}
+
+function device_manager.handle_sse_event(driver, device, event_type, data)
+  local _, device_json = pcall(json.decode, data)
+
+  local event_handler = sse_event_handlers[event_type]
+  if event_handler and device_json then
+    event_handler(driver, device, device_json)
+  else
+    log.error(string.format("handle_sse_event : unknown event type. dni = %s, event_type = '%s'",
+      device.device_network_id, event_type))
+  end
+end
+
+function device_manager.refresh(driver, device)
```

Comment:
This function is not needed. `device_manager.update_status` should just be called directly.